clippy_utils/src/msrvs.rs

```diff
@@ -159,34 +173,55 @@ pub struct MsrvStack {
 }
 
 impl MsrvStack {
+    /// Creates a new `MsrvStack` with the initial MSRV value from configuration
     pub fn new(initial: Msrv) -> Self {
         Self {
             stack: SmallVec::from_iter(initial.0),
         }
     }
 
+    /// Returns the MSRV at the current stack level
+    ///
+    /// This represents the effective MSRV after applying any `#[clippy::msrv]` attributes
+    /// encountered during traversal
     pub fn current(&self) -> Option<RustcVersion> {
         self.stack.last().copied()
     }
 
+    /// Checks if the current MSRV meets or exceeds the required version
+    ///
+    /// Returns `true` if no MSRV is configured or if the current MSRV is greater than
+    /// or equal to the required version
     pub fn meets(&self, required: RustcVersion) -> bool {
         self.current().is_none_or(|msrv| msrv >= required)
     }
 
+    /// Processes attributes to update the MSRV stack when entering a new scope
+    ///
+    /// If a `#[clippy::msrv]` attribute is found, parses the version and pushes it
+    /// onto the stack. Also marks that MSRV attributes have been seen globally.
     pub fn check_attributes(&mut self, sess: &Session, attrs: &[Attribute]) {
         if let Some(version) = parse_attrs(sess, attrs) {
             SEEN_MSRV_ATTR.store(true, Ordering::Relaxed);
             self.stack.push(version);
         }
     }
 
+    /// Processes attributes to update the MSRV stack when leaving a scope
+    ///
+    /// If a `#[clippy::msrv]` attribute is found, pops the most recently pushed
+    /// MSRV from the stack, effectively restoring the previous MSRV level
```

Comment:
Can be shortened to just "If the attributes contain a version attribute, pops a version from the stack."

A second paragraph about this being intended to be used from a lint pass's `check_attributes_post` function is also worth having.